config.go

```diff
@@ -0,0 +1,284 @@
+	return &cli.Command{
+		Name:        "config",
+		Usage:       "Manage app configurations",
+		Description: `Manage app configurations`,
+		Subcommands: []*cli.Command{
+			newConfigCmd(config),
+			removeConfigCmd(config),
+			listConfigsCmd(config),
+			defaultConfigCmd(config),
+		},
+	}
+}
+
+func newConfigCmd(config *Config) *cli.Command {
+	return &cli.Command{
+		Name:        "new",
+		Usage:       "Add a new App configuration",
+		UsageText:   "stream-cli config new",
+		Description: "Add a new App configuration which can be used on further operations",
+
+		Action: func(ctx *cli.Context) error {
+			newConfig := newDefaultConfig()
+			err := survey.Ask(questions(), &newConfig)
+			if err != nil {
+				return err
+			}
+
+			return config.Add(newConfig)
+		},
+	}
+}
+
+func removeConfigCmd(config *Config) *cli.Command {
+	return &cli.Command{
+		Name:        "remove",
+		Usage:       "Remove an App configuration",
+		UsageText:   "stream-cli config remove <app to remove>",
+		Description: "Remove an App configuration. This operation is irrevocable",
+
+		Action: func(ctx *cli.Context) error {
+			if ctx.Args().Len() != 1 {
+				return errors.New("remove command requires 1 argument")
+			}
+			return config.Remove(ctx.Args().First())
+		},
+	}
+}
+
+func listConfigsCmd(config *Config) *cli.Command {
+	return &cli.Command{
+		Name:        "list",
+		Usage:       "List all configurations",
+		UsageText:   "stream-cli config list",
+		Description: "List all app configurations",
+
+		Action: func(_ *cli.Context) error {
+			t := tabby.New()
+			t.AddHeader("", "Name", "Access Key", "Secret Key", "Region")
+
+			for k, v := range config.appsConfig {
+				defaultApp := ""
+				if v.Default {
+					defaultApp = "(default)"
+				}
+
+				secret := fmt.Sprintf("**************%v", v.AccessSecretKey[len(v.AccessSecretKey)-4:])
+				t.AddLine(defaultApp, k, v.AccessKey, secret, v.URL)
+			}
+			t.Print()
+			return nil
+		},
+	}
+}
+
+func defaultConfigCmd(config *Config) *cli.Command {
+	cmd := &cli.Command{
+		Name:        "default",
+		Usage:       "Set a configuration as the default",
+		UsageText:   "stream-cli config default <name of the configuration>",
+		Description: "Set a configuration as the default, it will be used by default for all operations",
+
+		Action: func(ctx *cli.Context) error {
+			if ctx.Args().Len() != 1 {
+				return errors.New("default command requires 1 argument")
+			}
+			return config.SetDefault(ctx.Args().First())
+		},
+	}
+
+	return cmd
+}
+
+type appConfig struct {
+	Name            string `yaml:"-"`
+	AccessKey       string `yaml:"access-key"`
+	AccessSecretKey string `yaml:"access-secret-key"`
+	URL             string `yaml:"url"`
+	Default         bool   `yaml:"default,omitempty"`
+}
+
+func newDefaultConfig() appConfig {
+	return appConfig{
+		URL: defaultEdgeURL,
+	}
+}
+
+type Config struct {
+	appsConfig map[string]*appConfig
+	file       *os.File
+}
+
+func NewConfig() (*Config, error) {
```

Comment:
Should we support custom config location ? like `--config ~/secret-cfg.yaml`.

Reply:
Yes we could but i leave this for a next iteration